### DISCUSSION TITLE
Does useContext in a page break static exports?

### DISCUSSION THREAD

**Discussion by asmenezes** [vercel/next.js#54993] [Help] [answered]
### Summary

When I use useContext in a component that gets used by a component in the "pages" folder the static export always fails with the following error (or something very similar)

Error occurred prerendering page "/initial". Read more: https://nextjs.org/docs/messages/prerender-error
TypeError: Cannot destructure property 'state' of '(0 , external_react_.useContext)(...)' as it is undefined.
    at ComponentA

Everything works fine if I run dev or start, only the static export is influenced, and only if the component is used either directly or indirectly by a "page", it exports just fine if I useContext in the default page.js. Is there some sort of NextJS specific thing that prevents this?

### Additional information

```js
Here is the component that when included in a "page" causes the static export to fail

"use client"
import React, { Fragment, useEffect, useContext, useState } from 'react'

import { MyContext } from './mycontext'
export default function ComponentA() {
    const {state , dispatch} =  useContext(MyContext)

return(
    <Fragment>A</Fragment>
)
}

I made a stack overflow post about this a week ago and didn't get any responses.  If for some reason the detail I go into there would be helpful here is the link.
https://stackoverflow.com/questions/76984483/usecontext-prevents-static-exports-in-nextjs-when-using-the-pages-folder
```

**Comment by minemos** [comment]
I have same error when enabled server actions

**Comment by joaopedrodcf** [comment, marked as answer]
For me it seems that you are calling the useContext but the parent component of this is not wrapped around a Provider. Because the useContext you called returned undefined and it failed because you tried to access a property that was not there called state.

I'm also using static exports with some contexts and it's working.

**Reply by asmenezes** [reply]
Yep its this, I was confused because when it runs on a node server the hierarchy goes Page>Initial>componentA which does have the context provider. But for the static export, it was exporting the initial page as a stand alone meaning Initial>ComponentA which does not have the provider since it does not go through Page, so I had to remove Page and just use Initial with the provider. Thanks for your help.

**Comment by icyJoseph** [comment]
Does it work if you move your `pages` folder into the `src` folder, I guess you have to rename it into something else? 

Otherwise could you make a stackblitz example where the error happens? https://stackblitz.com/

**Reply by asmenezes** [reply]
I did try moving things around when I saw your suggestion, but since I needed the pages to export statically moving and renaming the folder broke static routing. Another answer did help me out, thank you for your response.
